lighthouse-core/audits/seo/robots-txt.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const URL = require('../../lib/url-shim');
+
+const HTTP_CLIENT_ERROR_CODE_LOW = 400;
+const HTTP_SERVER_ERROR_CODE_LOW = 500;
+
+const DIRECTIVE_SITEMAP = 'sitemap';
+const DIRECTIVE_USER_AGENT = 'user-agent';
+const DIRECTIVE_ALLOW = 'allow';
+const DIRECTIVE_DISALLOW = 'disallow';
+const DIRECTIVES_GROUP_MEMBERS = new Set([DIRECTIVE_ALLOW, DIRECTIVE_DISALLOW]);
+const DIRECTIVE_SAFELIST = new Set([
+  DIRECTIVE_USER_AGENT, DIRECTIVE_DISALLOW, // standard
+  DIRECTIVE_ALLOW, DIRECTIVE_SITEMAP, // universally supported
+  'crawl-delay', // yahoo, bing, yandex
+  'clean-param', 'host', // yandex
+  'request-rate', 'visit-time', 'noindex', // not officially supported, but used in the wild
+]);
+const SITEMAP_VALID_PROTOCOLS = new Set(['https:', 'http:', 'ftp:']);
+
+/**
+ * @param {!string} directiveName
```

Comment:
no `!` necessary with our tsc typechecking anymore :)

mind adding this file to the typechecked files in `tsconfig` while we're at it?